File: PlusOne.java
class Solution {
    public int[] plusOne(int[] digits) {
        for(int i = digits.length - 1 ; i>=0 ; i--){
            digits[i] = (digits[i] + 1)%10;
            
            if(digits[i] != 0){
                return digits;
            }
        }
        int[] nums = new int[digits.length+1];
        nums[0] = 1;
        return nums;
    }
}

File: PlusOne.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    let carry = 1;
    for(let i=digits.length - 1; i>=0; i--) {
        const sum = digits[i] + carry;
        if(Math.floor(sum / 10) !== 0) {
            digits[i] = 0;
            carry = 1;
        }
        else {
            digits[i] = sum;
            return digits;
        }
    }
    return [1,...digits];
};
